app/src/main/java/org/glucosio/android/presenter/AddGlucosePresenter.java

```diff
@@ -119,6 +132,20 @@ public void dialogOnAddButtonPressed(String time, String date, String reading, S
         }
     }
 
+    @Nullable
+    private Number parseReading(@Nullable String reading) {
+        if (reading == null)
+            return null;
+        NumberFormat numberFormat = NumberFormat.getInstance();
+        Number number = null;
+        try {
+            number = numberFormat.parse(reading);
+        } catch (ParseException e) {
+            activity.showErrorMessage();
```

Comment:
You will show in this case error message two times - one here and another immediate after when null arrives to the calling method.